### PR TITLE
Added post_x_bindings methods for Python, Go and Julia bindings

### LINKED ISSUE #2581: Refactor CMake binding configuration
### What is the desired addition or change?

This comes out of the discussion for the R bindings: https://github.com/mlpack/mlpack/pull/2556#discussion_r470272755

At the end of `src/mlpack/CMakeLists.txt`, there is a section for each binding type.  It would be cleaner if we instead created macros like `post_markdown_setup()` for each of the existing binding languages (Julia, Go, Python) in each binding directory's `CMakeLists.txt` (i.e. `src/mlpack/bindings/*/CMakeLists.txt`), and then simply called those macros.

### What is the motivation for this feature?

A cleaner CMake configuration. :)

### If applicable, describe how this feature would be implemented.

So, in an ideal world, there would be a block at the end of `src/mlpack/CMakeLists.txt` like this:

```
post_python_setup()
post_julia_setup()
post_go_setup()
post_markdown_setup()
```

and that would be the only code in that file related to the bindings.

### Additional information?

This is a pretty simple refactoring, so it could be a nice starter task for anyone looking to get involved with mlpack.  Some CMake knowledge is probably necessary, but CMake isn't too hard to learn. :)

### PR DESCRIPTION
**Pull request by mikkac**
Second approach to resolve #2581 
First can be found in #2587- closed due to misunderstanding of the task

### CODE REVIEW

File: src/mlpack/bindings/go/CMakeLists.txt [hunk 1]
```diff
@@ -176,7 +200,7 @@ set_property(GLOBAL PROPERTY GO_MODELS "")
 macro (add_go_binding name)
 if (BUILD_GO_BINDINGS)
 
-  # Include all .h that define model to models.go. 
+  # Include all .h that define model to models.go.
```
Comment (rcurtin) [inline src/mlpack/bindings/go/CMakeLists.txt:203]: Thanks!  Always good to remove dangling whitespace. :+1:

File: src/mlpack/bindings/julia/CMakeLists.txt [hunk 2]
```diff
@@ -7,6 +7,16 @@ macro (not_found_return message)
   return()
 endmacro ()
 
+macro (post_julia_setup)
+  # If we are building Julia bindings, we have to end the 'module' declaration in
+  # mlpack.jl
```
Comment (rcurtin) [inline src/mlpack/bindings/julia/CMakeLists.txt:12]:
```suggestion
  # mlpack.jl.
```
Sorry for the pedantry. :smile:

File: src/mlpack/bindings/go/CMakeLists.txt [hunk 3]
```diff
@@ -9,6 +9,30 @@ macro (not_found_return message)
   endif()
 endmacro ()
 
+macro (post_go_setup)
+  if (BUILD_GO_BINDINGS)
+    # If we are building Go bindings, we have to end the 'module' declaration in
+    # models.go
```
Comment (rcurtin) [inline src/mlpack/bindings/go/CMakeLists.txt:15]:
```suggestion
    # Once `GO_MODELS` is populated, generate `models.go`.
```

Is this right?  I don't think that that there is any `module` that needs to be closed.  @Yashwants19, I think you originally wrote this; does my change look okay?  Let me know what you think. :+1:

**Review by rcurtin — COMMENTED** [review]
@mikkac thanks so much for looking into this!  It looks great to me.  I only have two really simple comments. :+1:

File: src/mlpack/bindings/go/CMakeLists.txt (hunk 3 above)
Comment (Yashwants19) [reply src/mlpack/bindings/go/CMakeLists.txt:15]: This is looking awesome, Thanks for improving the documentation. :)

**Review by rcurtin — APPROVED** [review]
Thanks @mikkac! :+1:

**Comment by rcurtin** [comment]
The failure appears unrelated, so I'll go ahead and merge.  Thanks @mikkac! :100:

**Pull request merged**